runtime/common/src/auctions.rs

```diff
@@ -1507,39 +1463,16 @@ mod tests {
 	#[test]
 	fn handle_bid_checks_existing_lease_periods() {
 		new_test_ext().execute_with(|| {
-			run_to_block(1);
-			assert_ok!(Auctions::new_auction(RuntimeOrigin::signed(6), 5, 1));
-			assert_ok!(Auctions::bid(RuntimeOrigin::signed(1), 0.into(), 1, 2, 3, 1));
-			assert_eq!(Balances::reserved_balance(1), 1);
-			assert_eq!(Balances::free_balance(1), 9);
-			run_to_block(9);
+			MockLeaser::mock_lease_period_index(|_| Some((0, 0))); // Used by new_auction
+			MockLeaser::mock_already_leased(|_, _, _| true); // Used by bid
 
-			assert_eq!(
-				leases(),
-				vec![
-					((0.into(), 2), LeaseData { leaser: 1, amount: 1 }),
-					((0.into(), 3), LeaseData { leaser: 1, amount: 1 }),
-				]
-			);
-			assert_eq!(TestLeaser::deposit_held(0.into(), &1), 1);
+			run_to_block(1);
 
-			// Para 1 just won an auction above and won some lease periods.
-			// No bids can work which overlap these periods.
 			assert_ok!(Auctions::new_auction(RuntimeOrigin::signed(6), 5, 1));
 			assert_noop!(
 				Auctions::bid(RuntimeOrigin::signed(1), 0.into(), 2, 1, 4, 1),
 				Error::<Test>::AlreadyLeasedOut,
 			);
-			assert_noop!(
-				Auctions::bid(RuntimeOrigin::signed(1), 0.into(), 2, 1, 2, 1),
-				Error::<Test>::AlreadyLeasedOut,
-			);
-			assert_noop!(
-				Auctions::bid(RuntimeOrigin::signed(1), 0.into(), 2, 3, 4, 1),
-				Error::<Test>::AlreadyLeasedOut,
-			);
-			// This is okay, not an overlapping bid.
-			assert_ok!(Auctions::bid(RuntimeOrigin::signed(1), 0.into(), 2, 1, 1, 1));
 		});
 	}
```

Comment:
If I understand this method well, it can be simplified as the diff shows.

The purpose of this test is to check if the `bid()` method correctly checks the value returned by `already_leased()`. Ideally, it does not need more than checking the boolean that the method returns. The three removed `assert_noop()` are checking that `TestLeaser::already_leased()` is working well in different circumstances, but we do not want to check the `TestLeaser` implementation, we want to check the pallet implementation.

Mocking the exact method allows us to split the real code from the testing code, letting us have smaller and simpler test cases.